packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -0,0 +1,190 @@
+
+import { Title } from '../../components/Head';
+import Button from '../Button';
+import PriceSummary from './PriceSummary';
+import PriceAdjustments from './PriceAdjustments';
+import PaymentInformation from './PaymentInformation';
+import ShippingMethod from './ShippingMethod';
+import ShippingInformation from './ShippingInformation';
+
+import { renderIf } from './utilities';
+import { mergeClasses } from '../../classify';
+
+import defaultClasses from './checkoutPage.css';
+
+const GuestCheckoutOptions = props => {
+    const { classes, handleSignIn } = props;
+
+    return (
+        <div className={classes.signin_container}>
+            <div className={classes.heading_container}>
+                <h1 className={classes.heading}>Quick Checkout</h1>
+            </div>
+            <Button
+                className={classes.sign_in}
+                onClick={handleSignIn}
+                priority="high"
+            >
+                {'Login to Checkout Faster'}
+            </Button>
+        </div>
+    );
+};
+
+const GreetingMessage = props => {
+    const { isGuestCheckout, classes } = props;
+
+    return (
+        <div className={classes.heading_container}>
+            <h1 className={classes.heading}>
+                {isGuestCheckout ? 'Guest Checkout' : 'Review and Place Order'}
+            </h1>
+        </div>
+    );
+};
+
+const EmptyCartMessage = props => {
+    const { classes, isGuestCheckout } = props;
+
+    return (
+        <div className={classes.signin_container}>
+            <div className={classes.heading_container}>
+                <h1 className={classes.heading}>
+                    {isGuestCheckout ? 'Guest Checkout' : 'Checkout'}
+                </h1>
+            </div>
+            <h3>There are no items in your cart.</h3>
+        </div>
+    );
+};
+
+export default props => {
+    const { classes: propClasses } = props;
+    const [
+        {
+            isGuestCheckout,
+            isCartEmpty,
+            shippingInformationDone,
+            shippingMethodDone
+        },
+        { handleSignIn, setShippingInformationDone, setShippingMethodDone }
+    ] = useCheckoutPage();
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const renderIfGuestCheckout = useCallback(renderIf(isGuestCheckout), [
+        isGuestCheckout
+    ]);
+    const renderIfCartNotEmpty = useCallback(renderIf(!isCartEmpty), [
+        isCartEmpty
+    ]);
+    const renderIfShippingInformationDone = useCallback(
+        renderIf(shippingInformationDone)
+    );
+    const renderIfShippingMethodDone = useCallback(
+        renderIf(shippingMethodDone)
+    );
+
+    return (
+        <div className={classes.root}>
+            <Title>{`Checkout - ${STORE_NAME}`}</Title>
+            {/**
+             * Lazy rendering checkout page components only
+             * if the cart is not empty.
+             */
+            renderIfCartNotEmpty(
```

Comment:
Instead of using this conditional render, this should be a lazy query, triggered off `cart_id`, for `total_quantity`. Check out how `CartPage` does it.

Reply:
I thought of leaving that to the individual taking up the respective checkout tickets.

Reply:
Yea, I just thought this was relevant to the initial bare bone state :P Regardless, made a ticket to follow up.